transaction_essence.go

```diff
@@ -286,32 +321,23 @@ func (u *TransactionEssence) UnmarshalJSON(bytes []byte) error {
 	return nil
 }
 
-// SyntacticallyValidate checks whether the transaction essence is syntactically valid by checking whether:
-//	1. every input references a unique UTXO and has valid UTXO index bounds
-//	2. every output (per type) deposits to a unique address and deposits more than zero
-//	3. the accumulated deposit output is not over the total supply
-//	4. SigLockedDustAllowanceOutput deposits at least OutputSigLockedDustAllowanceOutputMinDeposit.
+// syntacticallyValidate checks whether the transaction essence is syntactically valid.
 // The function does not syntactically validate the input or outputs themselves.
-func (u *TransactionEssence) SyntacticallyValidate() error {
-
-	if len(u.Inputs) == 0 {
-		return ErrMinInputsNotReached
-	}
-
-	if len(u.Outputs) == 0 {
-		return ErrMinOutputsNotReached
-	}
-
-	if err := ValidateInputs(u.Inputs,
-		InputsUTXORefIndexBoundsValidator(),
-		InputsUTXORefsUniqueValidator(),
+func (u *TransactionEssence) syntacticallyValidate(protoParas *ProtocolParameters) error {
```

Comment:
This is missing the `Network ID must match the value of the current network.` syntactical validation comparing it to the given ProtocolParameters